struct_test.go

```diff
@@ -472,6 +472,24 @@ func TestRegisterFunc_structArgs(t *testing.T) {
 			t.Fatalf("TakeGoUintAndReturn returned %+v wanted %+v", ret, expected)
 		}
 	}
+	{
+		// Regression test for bool field at non-zero offset
+		// Before fix: bool was always placed at bit 0 regardless of offset
+		// After fix: bool is correctly shifted to its field offset
+		type FloatAndBool struct {
+			Value     float32
+			Has_value bool
```

Comment:
Uncommon to use underscores in Go fields. Call it `HasValue`? doesn't matter if it matches the C name exactly